pkg/server/server.go

```diff
@@ -2213,6 +2219,29 @@ func (s *topLevelServer) PreStart(ctx context.Context) error {
 	return maybeImportTS(ctx, s)
 }
 
+// initJobScheduler starts the job scheduler. This must be called
+// after sqlServer.preStart and after our external storage providers
+// have been initialized.
+//
+// TODO(ssd): We need to clean up the ordering/ownership here. The SQL
+// server owns the job scheduler because the job scheduler needs an
+// internal executor. But, the topLevelServer owns initialization of
+// the external storage providers.
+func (s *topLevelServer) initJobScheduler(ctx context.Context) error {
+	if s.cfg.DisableSQLServer {
+		return nil
+	}
+	// The job scheduler may immediately start jobs that require
+	// external storage providers to be available. We expect the
+	// server start up ordering to ensure this. Hitting this error
+	// is a programming error somewhere in server startup.
+	if err := s.externalStorageBuilder.assertInitComplete(); err != nil {
+		return err
+	}
+	s.sqlServer.startJobScheduler(ctx, s.cfg.TestingKnobs)
```

Comment:
nit: is this to many layers of indirection? This init method is not really init'ing anything so maybe we can just move this logic to startJobScheduler? I don't feel strongly so feel free to ignore.

Reply:
I kinda agree, but to me its kinda ugly either way. Having a function that just calls another function is definitely a bit meh. But then if we move that call inline here we suddenly have a number of more places where we reach into sqlServer's fields directly.